### PR TITLE
Upgraded ElasticSearch to get rid of CVEs.

### PR DESCRIPTION
**Pull request by dlg99**
CVEs are:
CVE-2020-7020
CVE-2020-7021
CVE-2021-22132
CVE-2021-22134
CVE-2021-22144
CVE-2021-22147

### Motivation

`mvn clean install verify -Powasp-dependency-check -DskipTests` found various CVEs

### Modifications

Upgraded ElasticSearch to get rid of CVEs.
Had to update maven compiler plugin config, otherwise it was NPE-ing instead of showing actual errors.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): *YES*
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by nicoloboschi — APPROVED** [review]

**Review by lhotari — CHANGES_REQUESTED** [review]
> Had to update maven compiler plugin config, otherwise it was NPE-ing instead of showing actual errors.

Please revert these changes from this PR, if possible. 

There's #13126 about the maven NPE issue. In the past there has been a need to temporarily set `forceJavacCompilerUse` to find out the compiler error, but it has been possible to remove the option after fixing the compilation issue. Is that also the case here?

At least it seems that setting the maven compiler plugin version isn't necessary since the parent pom sets it.

If there's no way around it, then add a comment why `forceJavacCompilerUse` has been added, such as.
```xml
         <!-- workaround https://issues.apache.org/jira/browse/MCOMPILER-346 -->
         <forceJavacCompilerUse>true</forceJavacCompilerUse>
```

**Comment by dlg99** [comment]
@lhotari I removed forceJavacCompilerUse and everything builds/pushed the change. 
Please help me understand what's the downside of just keeping it there/adding in another PR?
I presume everyone who updates dependencies now and then gets into that NPE in maven, and have to add the forceJavacCompilerUse to see the errors, and remove it later. 
So does it hurt build times or anything else? Why don't we make developers' life easier and keep it until we move to the maven/jvm plugin with the [NPE fixed]( https://bugs.openjdk.java.net/browse/JDK-8216202) (not soon, probably, as it is not fixed yet)?

File: pulsar-io/elastic-search/pom.xml [hunk 1]
```diff
@@ -83,7 +83,7 @@
     <dependency>
       <groupId>org.testcontainers</groupId>
       <artifactId>elasticsearch</artifactId>
-      <version>1.15.3</version>
+      <version>1.16.2</version>
```
Comment (lhotari) [inline pulsar-io/elastic-search/pom.xml:86]: Please remove the version since the root pom.xml imports testcontainers-bom .

**Review by lhotari — CHANGES_REQUESTED** [review]

**Comment by lhotari** [comment]
> @lhotari I removed forceJavacCompilerUse and everything builds/pushed the change. Please help me understand what's the downside of just keeping it there/adding in another PR? I presume everyone who updates dependencies now and then gets into that NPE in maven, and have to add the forceJavacCompilerUse to see the errors, and remove it later. So does it hurt build times or anything else? Why don't we make developers' life easier and keep it until we move to the maven/jvm plugin with the [NPE fixed](https://bugs.openjdk.java.net/browse/JDK-8216202) (not soon, probably, as it is not fixed yet)?

I believe that the downside is reduced performance. In addition, the problem might happen in any module and therefore it doesn't make sense to change the setting for just a few modules if it were to be enabled globally.
There might also be some changes in how maven compiler issues are displayed when forceJavacCompilerUse is used.

The way to make developers' life easier is to raise awareness of this issue. I added a simpler workaround to #13126, it's by passing `-Dmaven.compiler.forceJavacCompilerUse=true` to the mvn command line. To make it easier to find this information we could add the information to README or the contributor guide.

**Comment by lhotari** [comment]
btw. Regarding the NPEs with maven-compiler-plugin, version 3.9.0 was released recently with fixes to some NPE issues. https://lists.apache.org/thread/8bm3powmfy46z25k9jgrbkf1kxf5j6yk . I'll create a PR to upgrade to 3.9.0 .

**Review by lhotari — APPROVED** [review]
Good work @dlg99 ! LGTM

**Review by eolivelli — APPROVED** [review]
LGTM

great

**Review by Jason918 — APPROVED** [review]

**Pull request merged**